lib/minitest/stub_const.rb

```diff
@@ -37,15 +36,21 @@ def stub_const(name, val, &block)
   # Example:
   #
   #   Object.stub_remove_const(:File) do
-  #     "Look ma, no File!" unless defined(File)
+  #     "Look ma, no File!" unless defined?(File)
   #   end
   #   # => "Look ma, no File!"
   def stub_remove_const(name)
-    orig = const_get(name)
-    remove_const(name)
-    yield
-  ensure
-    const_set(name, orig)
+    if const_defined?(name)
+      begin
+        orig = const_get(name)
+        remove_const(name)
+        yield
+      ensure
+        const_set(name, orig)
+      end
+    else
+      yield
```

Comment:
I see your point; I am not quite sure if this is the best way to handle this type of situation myself.

As to the use case: This method was the reason I submitted the pull request. I have a project where I have a soft dependency on mustache and use pencil_mustache as a fallback. In order to be able to test the fallback, I `stub_remove_const` the `Mustache` constant. However, currently this only works if the mustache gem is actually installed. (To make matters worse, Mustache does not support Ruby <2.0 anymore.)

Basically this change would make it easier to drop development dependencies that are only included for the sake of some constant being available at the test's runtime without having to duplicate every `stub_remove_const` block because of `const_defined?` checks.